docs/userguide/icu4j/index.md

```diff
@@ -68,41 +67,122 @@ There are a few different ways to download the ICU4J releases.
+| `main/translit`   | `translit`     | `core` | `com.ibm.icu.text` | Transliterator APIs and implementation. This sub-component is packaged as a part of `icu4j.jar`. |
+
+### ICU4J unit test files
+
+Unit test and integration tests are grouped together in the same Maven submodule as the runtime class code
+for the same component.
+The test source code lives in the standard Maven directory structure,
+`src/test/java` for test sources,
+and `src/test/resources` for test-specific non-source files.
+
+> :point_right: **Note**: Tests that depend on multiple sub-components must be placed
+in the `main/common_tests` directory and run as integration tests.
+
+The Maven build enforces isolation vis-a-vis dependencies for tests within a multi-module project such that
+unit tests must only test code within its same submodule.
+When a test depends on runtime code from multiple modules, it cannot exist in any of those modules,
+or else it would create a circular dependency in the Maven build.
+Instead, such tests should be refactored into a separate component
+and designated as integration tests.
+
+| Path                                      | Sub-component Name              | Runtime Dependencies                                                                                         | Description                                     |
+|-------------------------------------------|---------------------------------|--------------------------------------------------------------------------------------------------------------|-------------------------------------------------|
+| `main/common_tests`                        | `common_tests`             | `core` <br/> `currdata` <br/> `translit` <br/> `langdata` <br/> `collate` <br/> `regiondata` <br/>                                                                 | Sub-component for integration tests that depend on runtime code of multiple components. |
+
+
+### Others
+
+<table>
+    <tr>
+        <th>Path</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td><code>main/shared</code></td>
+        <td>(Obsolete?) Files shared by ICU4J sub-components under the <code>main</code> directory including: <ul> <li>ICU4J runtime data archive (icudata.jar).</li> <li>ICU4J unit test data archive (testdata.jar).</li> <li>Shared Ant build script and configuration files.</li> <li>License files.</li></ul> </td>
+    </tr>
+    <tr>
+        <td><code>demos</code></td>
+        <td>ICU4J demo programs.</td>
+    </tr>
+    <tr>
+        <td><code>perf-tests</code></td>
+        <td>ICU4J performance test files.</td>
+    </tr>
+    <tr>
+        <td><code>tools</code></td>
+        <td>ICU4J tools including: <ul> <li>Custom JavaDoc taglets used for generating ICU4J API references.</li> <li>API report tool and data.</li> <li>Other independent utilities used for ICU4J development.</li> </ul></td>
+    </tr>
+</table>
+
+## The Structure and Contents of ICU4J - ICU 73 and earlier
```

Comment:
Idea: split in a different file, to reduce the clutter?
And with only a link from here?

Reply:
Markus wanted to keep the Ant instructions for people who might still be using a version of ICU4J that is a few versions old. There will be a non-trivial number of people.